### PR TITLE
Huh

### PR DESCRIPTION
**Pull request by GurteX**
Why is it like that

### CODE REVIEW

**Review by cattyngmd — CHANGES_REQUESTED** [review]
it would be better if u check for blocks hardness instead

**Comment by 3arthqu4ke** [comment]
See Speedmine

**Comment by GurteX** [comment]
Wow its a quick fix man, just this

**Comment by 3arthqu4ke** [comment]
Nono this Enterprise Software has no place for, to use your words "quick fixes"! 🤓

**Comment by 3arthqu4ke** [comment]
Also U get the blockstate twice here

**Comment by GurteX** [comment]
> Also U get the blockstate twice here

Bad bait considering that you didnt think of this b4

File: src/main/java/me/earth/phobot/services/BlockDestructionService.java [hunk 1]
```diff
@@ -30,8 +30,10 @@ public BlockDestructionService(Minecraft mc) {
         listen(new ReceiveListener.Scheduled.Safe<ClientboundBlockDestructionPacket>(mc) {
             @Override
             public void onSafeEvent(PacketEvent.Receive<ClientboundBlockDestructionPacket> event, LocalPlayer self, ClientLevel level, MultiPlayerGameMode gameMode) {
-                if (!level.getBlockState(event.getPacket().getPos()).isAir()) {
-                    addProgress(event.getPacket().getPos(), level);
+                BlockPos p = event.getPacket().getPos();
+                if (canRecord(p, level)) { // should prevent rendering bedrock??
+                    addProgress(p, level);
+                    PingBypassApi.getEventBus().post(new FriendStartBlockMineEvent(p));
```
Comment (3arthqu4ke) [inline src/main/java/me/earth/phobot/services/BlockDestructionService.java]: What is a FriendStartBlockMineEvent? This wont compile.

File: src/main/java/me/earth/phobot/services/BlockDestructionService.java [hunk 2]
```diff
@@ -73,4 +75,9 @@ public boolean isInvalid() {
         }
     }
 
+    private boolean canRecord(BlockPos pos, Level level) {
+        BlockState state = level.getBlockState(pos);
+        return state.getDestroySpeed(level, pos) < 0 && !state.isAir();
```
Comment (3arthqu4ke) [inline src/main/java/me/earth/phobot/services/BlockDestructionService.java]: Condition is wrong destroyspeed < 0 means the Block is Not breakable.

**Review by 3arthqu4ke — CHANGES_REQUESTED** [review]

**Comment by GurteX** [comment]
oops lol i didnt see that other event my bad g

**Comment by GurteX** [comment]
im never writing shi on github instead of an ide again

File: src/main/java/me/earth/phobot/services/BlockDestructionService.java [hunk 3]
```diff
@@ -73,4 +74,9 @@ public boolean isInvalid() {
         }
     }
 
+    private boolean canRecord(BlockPos pos, Level level) {
+        BlockState state = level.getBlockState(pos);
+        return state.getDestroySpeed(level, pos) > 0 && !state.isAir();
```
Comment (3arthqu4ke) [inline src/main/java/me/earth/phobot/services/BlockDestructionService.java]:
```java
return state.getDestroySpeed(level, pos) >= 0 && !state.isAir();
```

**Review by 3arthqu4ke — CHANGES_REQUESTED** [review]

File: src/main/java/me/earth/phobot/services/BlockDestructionService.java (hunk 3 above)
Comment (3arthqu4ke) [reply src/main/java/me/earth/phobot/services/BlockDestructionService.java]: Also fix build

Comment (GurteX) [reply src/main/java/me/earth/phobot/services/BlockDestructionService.java]: > ```java
> return state.getDestroySpeed(level, pos) >= 0 && !state.isAir();
> ```

Is 0 a valid destroy speed

Comment (GurteX) [reply src/main/java/me/earth/phobot/services/BlockDestructionService.java]: > Also fix build

what build do you want me to fix ???

**Comment by GurteX** [comment]
mf went on dem holidays neva coming back

**Comment by GurteX** [comment]
Bro

**Review by 3arthqu4ke — APPROVED** [review]
You did it!

**Comment by GurteX** [comment]
> You did it!

Real

**Pull request merged**